app/controllers/save_pages_controller.rb

```diff
@@ -0,0 +1,28 @@
+class SavePagesController < ApplicationController
+  include AccountConcern
+
+  before_action :set_no_cache_headers
+  before_action :authenticate
+
+  def create
+    redirect_to GdsApi.account_api.save_page(
+      page_path: params[:page_path],
+      govuk_account_session: account_session_header,
+    ).to_h["saved_page"]["page_path"]
+  rescue GdsApi::HTTPUnauthorized
+    @account_session_header = nil
+    authenticate
+  end
```

Comment:
We'll need to handle the 422 case here, as currently that'll propagate upwards and the user will get a 422 response.

If we go for the query param approach, maybe we could do

```ruby
redirect_to params[:page_path] + "?personalisation=page_not_saved"
```

And display an error box